### PR TITLE
Separate MutableX and ProxyX

### PR DESCRIPTION
**Pull request by KernelDeimos**

### CODE REVIEW

File: android/nanos_example_client/src/main/java/com/foamdev/examples/nanos_client/MainActivity.java [hunk 1]
```diff
@@ -12,20 +12,20 @@ public class MainActivity extends android.app.ListActivity {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        foam.core.ProxyX proxyX = new foam.core.ProxyX();
+        foam.core.MutableX MutableX = new foam.core.MutableX();
```
Comment (kgrgreer) [inline android/nanos_example_client/src/main/java/com/foamdev/examples/nanos_client/MainActivity.java]: Should be named mutableX, not MutableX.

File: src/foam/core/AbstractAgency.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ foam.CLASS({
       if ( ! TASK_QUEUE.containsKey(key) ) {
         var task = new TimerTask() {
           public void run() {
-            X oldX = ((ProxyX) XLocator.get()).getX();
+            X oldX = ((MutableX) XLocator.get()).getX();
```
Comment (kgrgreer) [inline src/foam/core/AbstractAgency.js:39]: Would a ProxyX actually be better in this case?

File: src/foam/core/ContextAgentRunnable.java [hunk 3]
```diff
@@ -24,7 +24,7 @@ public String toString() {
   }
 
   public void run() {
-    X oldX = ((ProxyX) XLocator.get()).getX();
+    X oldX = ((MutableX) XLocator.get()).getX();
```
Comment (kgrgreer) [inline src/foam/core/ContextAgentRunnable.java:27]: Why ProxyX or MutableX, why not just X?

File: src/foam/core/MutableX.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+ */
+
+package foam.core;
+
+import java.util.Collections;
+import java.util.Map;
+
+/** Proxy for X interface. **/
+public class MutableX
+  extends    ContextAwareSupport
```
Comment (kgrgreer) [inline src/foam/core/MutableX.java]: Would be better to extend ProxyX.

File: src/foam/core/SubX.java [hunk 5]
```diff
@@ -10,7 +10,7 @@
 import java.util.HashSet;
 import java.util.Set;
 
-public class SubX extends ProxyX {
+public class SubX extends MutableX {
```
Comment (kgrgreer) [inline src/foam/core/SubX.java:13]: What does this class do?

File: src/foam/core/XLocator.java [hunk 6]
```diff
@@ -10,9 +10,9 @@
 public class XLocator
 {
 
-  protected static ThreadLocal<ProxyX> x__ = new ThreadLocal<ProxyX>() {
-    protected ProxyX initialValue() {
-      return new ProxyX();
+  protected static ThreadLocal<MutableX> x__ = new ThreadLocal<MutableX>() {
```
Comment (kgrgreer) [inline src/foam/core/XLocator.java:13]: I think better as just a Proxy (if it doesn't break anything).

**Comment by jlhughes** [comment]
@KernelDeimos the intent was to just use the MutableX in the boot sequence, it should be the only place it is needed.

File: src/foam/core/AbstractAgency.js (hunk 2 above)
Comment (KernelDeimos) [reply src/foam/core/AbstractAgency.js:39]: What do users of an `XLocator` usually expect? A mutable or non-mutable X?

**Pull request closed without merging**